src/client/AutherClient.js

```diff
@@ -56,4 +123,17 @@ export class AutherClient {
 
     return this.http({ path: this.#REFRESH_PATH, body: { refreshToken } })
   }
+
+  authentication = async ({ fetchTokens, saveTokens }) => {
+    const { accessToken } = fetchTokens()
+    try {
+      verify(accessToken)
+    }
+    catch (err) {
+      await this.#refresh({ fetchTokens, saveTokens, immediate: true })
+      return
+    }
+
+    this.#refresh({ fetchTokens, saveTokens, immediate: false })
+  }
```

Comment:
ну лично мне кажется, логика несколько переусложенена с этим immediate, не? можно же просто сделать 

```suggestion
  authentication = async ({ fetchTokens, saveTokens }) => {
    const { accessToken } = fetchTokens()
    try {
      verify(accessToken)
    }
    catch (err) {
      await this.#refresh({ fetchTokens, saveTokens })
      return
    }

    this.#scheduleRefreshTokens({ fetchTokens, saveTokens })
  }
```

Reply:
лично на мой вкус так более явно прослеживается намерение, чем прыгать по методам с иммедиейтом